core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -520,6 +520,10 @@ private[spark] class SparkSubmit extends Logging {
         confKey = "spark.driver.extraJavaOptions"),
       OptionAssigner(args.driverExtraLibraryPath, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
         confKey = "spark.driver.extraLibraryPath"),
+      OptionAssigner(args.principal, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
+        confKey = "spark.kerberos.principal"),
```

Comment:
Since you're changing this, could you use `PRINCIPAL.key` and `KEYTAB.key` in the places where the config name is referenced?